File: java/src/test/java/mx/iteso/WaterIceCreamStoreTest.java
package mx.iteso;

import static org.junit.Assert.assertEquals;
import org.junit.Test;
import mx.iteso.Cone;
import mx.iteso.stores.WaterIceCreamStore;

public class WaterIceCreamStoreTest {
    @Test
    public void waterIceCreamStoreGuavaTest() {
        Cone cone = new WaterIceCreamStore().prepareCone("Guava");
        assertEquals(cone.getIceCream().getFlavor(), "Agua -> Guayaba");
    }

    @Test
    public void waterIceCreamStoreMangoTest() {
        Cone cone = new WaterIceCreamStore().prepareCone("Mango");
        assertEquals(cone.getIceCream().getFlavor(), "Agua -> Mango");
    }

    @Test
    public void waterIceCreamStorePineappleTest() {
        Cone cone = new WaterIceCreamStore().prepareCone("Pineapple");
        assertEquals(cone.getIceCream().getFlavor(), "Agua -> Piña");
    }

    @Test
    public void waterIceCreamStoreStrawberryTest() {
        Cone cone = new WaterIceCreamStore().prepareCone("Strawberry");
        assertEquals(cone.getIceCream().getFlavor(), "Agua -> Fresa");
    }
}

File: python/src/unittest/python/WaterIceCreamStoreTest.py
from mockito import mock, verify, when
import unittest
from store.WaterIceCreamStore import *


class WaterIceCreamStoreTest(unittest.TestCase):

    def setUp(self):
        self.store = WaterIceCreamStore()

    def guavaWaterIceCreamTest(self):
        cone = self.store.prepareCone("Guava")
        self.assertEquals(cone.getIceCream().getFlavor(), "Agua -> Guayaba")

    def mangoWaterIceCreamTest(self):
        cone = self.store.prepareCone("Mango")
        self.assertEquals(cone.getIceCream().getFlavor(), "Agua -> Mango")

    def pineappleWaterIceCreamTest(self):
        cone = self.store.prepareCone("Pineapple")
        self.assertEquals(cone.getIceCream().getFlavor(), "Agua -> Piña")

    def strawberryWaterIceCreamTest(self):
        cone = self.store.prepareCone("Strawberry")
        self.assertEquals(cone.getIceCream().getFlavor(), "Agua -> Fresa")